nef/Extensions/AppDelegate+Notifications.swift

```diff
@@ -0,0 +1,102 @@
+//  Copyright © 2020 The nef Authors.
+
+import AppKit
+import UserNotifications
+import Bow
+import BowEffects
+
+extension AppDelegate: UNUserNotificationCenterDelegate {
+    func isLocalNotification(_ aNotification: Notification) -> Bool {
+        guard let userInfo = aNotification.userInfo,
+              let launchOption = userInfo["NSApplicationLaunchIsDefaultLaunchKey"] as? Int else { return false }
+        
+        return launchOption == 0
+    }
+    
+    func registerNotifications() {
+        notificationCenter.delegate = self
+        notificationCenter.requestAuthorization(options: [.alert, .sound]) { granted, _  in }
+    }
+    
+    func showNotification(title: String, body: String, imageData: Data? = nil, actions: [NefNotificationAction] = [], id: String = UUID().uuidString) {
+        let content = UNMutableNotificationContent()
+        content.title = title
+        content.body = body
+        content.categoryIdentifier = id
+        
+        if let data = imageData {
+            content.userInfo = [Self.imageDataUserInfoKey: data]
+        }
+        
+        let category = UNNotificationCategory(identifier: id,
+                                              actions: actions.map(\.unNotificationAction),
+                                              intentIdentifiers: [],
+                                              hiddenPreviewsBodyPlaceholder: "",
+                                              options: .customDismissAction)
+        
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 0.5, repeats: false)
+        let request = UNNotificationRequest(identifier: id, content: content, trigger: trigger)
+        
+        notificationCenter.removeAllDeliveredNotifications()
```

Comment:
what do you think extract this call to another method in the extension (ex. `removeOldNotifications`), in order to keep a single responsibility here. I want to show notifications (but it is also removing last notifications) - we could invoke it in the logic of `pasteboardCarbonDidFinishLaunching` wdut?